core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -219,13 +219,13 @@ public synchronized void connect() throws AlluxioStatusException {
         }
       }
       // TODO(peis): Consider closing the connection here as well.
-      if (!retryPolicy.attemptRetry()) {
+      if (!retryPolicy.attempt()) {
```

Comment:
I think we can remove this if we do the check in the while condition

Reply:
Done